### PR TITLE
Add Clipboard events

### PR DESCRIPTION
**Pull request by kirilldakhniuk**
This PR should add an abilty to fire Clipboard events. If this PR seems reasonable to you please leave a comment and I will create PR for [NativePHP Laravel Repo](https://github.com/NativePHP/) to handle these events. Thanks.

### CODE REVIEW

**Comment by simonhamp** [comment]
How do you intend to use this? Can you explain what you're doing/trying to do?

Also, I'm not sure what the name `proceedToNotification` is meant to imply... it seems ambiguous to me. Can we come up with a better name?

**Comment by kirilldakhniuk** [comment]
I was trying to build a clipboard manager. The idea was to listen for clipboard events on Livewire component so I can create a new DB record. Earlier I thought about `wire:poll` to check if clipboard has value, but it's not a good idea. 
Could not agree more on naming... But I would think about a better one or might you suggest an option. Surely if you think it's something worth adding.

File: src/server/api/enums/ClipboardContentTypes.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+enum ClipboardContentTypes {
```
Comment (simonhamp) [inline src/server/api/enums/ClipboardContentTypes.ts:1]: I'm not sure if this is standard practice in TypeScript world, but I prefer singular names for `enums`

```suggestion
enum ClipboardContentType {
```

File: src/server/api/enums/ClipboardEvents.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+enum ClipboardEvents {
```
Comment (simonhamp) [inline src/server/api/enums/ClipboardEvents.ts:1]:
```suggestion
enum ClipboardEvent {
```

File: src/server/api/enums/ClipboardEvents.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  Read = 'read',
+  Written = 'written'
```
Comment (simonhamp) [inline src/server/api/enums/ClipboardEvents.ts]:
```suggestion
  Write = 'write'
```

File: src/server/api/clipboard.ts [hunk 4]
```diff
@@ -82,4 +95,13 @@ router.delete('/', (req, res) => {
     res.sendStatus(200);
 });
 
+function proceedToNotification(eventType: ClipboardEvents, contentType: ClipboardContentTypes) {
```
Comment (simonhamp) [inline src/server/api/clipboard.ts]:
```suggestion
function forwardToLaravel(eventType: ClipboardEvents, contentType: ClipboardContentTypes) {
```

**Review by simonhamp — CHANGES_REQUESTED** [review]
Here are my suggestions

File: src/server/api/enums/ClipboardEvents.ts (hunk 2 above)
Comment (kirilldakhniuk) [reply src/server/api/enums/ClipboardEvents.ts:1]: @simonhamp seems like I cannot do that, what's your opinion on ClipboardActionEvent?

image

Comment (simonhamp) [reply src/server/api/enums/ClipboardEvents.ts:1]: If there's already a `ClipboardEvent` enum, can we just use that?

Comment (kirilldakhniuk) [reply src/server/api/enums/ClipboardEvents.ts:1]: I don't think it's an enum.

image

Comment (simonhamp) [reply src/server/api/enums/ClipboardEvents.ts:1]: Ah ok, let's stick with `ClipboardEvents` - sorry for the turnaround

File: src/server/api/enums/ClipboardContentTypes.ts (hunk 1 above)
Comment (kirilldakhniuk) [reply src/server/api/enums/ClipboardContentTypes.ts:1]: @simonhamp do you still want this enum to have a singular name even though ClipboardEvents stays plural?

**Comment by kirilldakhniuk** [comment]
@simonhamp I made changes according to our chat. However I keep enum names plural to be consistent. Let me know if you have any other suggestions. Thanks.

**Comment by simonhamp** [comment]
I've looked through this with a fresh pair of eyes and I have more concerns...

These clipboard endpoints are used _by_ the Laravel side of the application to read from and write to the clipboard... so in theory, if you're just trying to get event data over into Livewire, you should just be able to fire [a custom event](https://nativephp.com/docs/1/digging-deeper/broadcasting#broadcasting-custom-events) from your application that [Livewire should pick it up](https://nativephp.com/docs/1/digging-deeper/broadcasting#livewire) via the `native:` callsign.

That would prevent this possible infinite loop that I think this implementation could more easily create.

**Comment by simonhamp** [comment]
@kirilldakhniuk I'm going to close this for now, as I can't quite fathom how this PR is going to give you what you're after.

I did a bit of digging, and I think what you need [isn't available in Electron](https://github.com/electron/electron/issues/2280).

You could implement something like a polling mechanism (e.g. a Scheduled Task in Laravel which checked the clipboard on a regular basis), but I think this is not very performant and would be quite a nasty solution imo.

Sorry for the run-around.

**Comment by kirilldakhniuk** [comment]
@simonhamp got it. I thought about pulling using a command and felt the same unpleasant feeling about it. 
Thank you.

**Pull request closed without merging**